src/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -221,7 +224,39 @@ public override string ToString()
 
         public WebHeaderCollection()
         {
-        }     
+        }
+
+        protected WebHeaderCollection(SerializationInfo serializationInfo, StreamingContext streamingContext)
+        {
+            int count = serializationInfo.GetInt32("Count");
```

Comment:
I considered it, but as desktop doesn't have such a check, I thought it better to just have a null here result in the same exception type that shows up there.  This method isn't typically used directly by a caller, only by a formatter via reflection and by a derived type passing down the SerializationInfo and StreamingContext it was given.